tests/crypto/test_cases/test_aes_ecb_mac.c

```diff
@@ -0,0 +1,211 @@
+{
+	mbedtls_cipher_init(p_ctx);
+
+	const mbedtls_cipher_info_t *p_cipher_info =
+		mbedtls_cipher_info_from_values(MBEDTLS_CIPHER_ID_AES,
+						key_len_bytes * 8, mode);
+	return mbedtls_cipher_setup(p_ctx, p_cipher_info);
+}
+
+static int cipher_set_key(mbedtls_cipher_context_t *p_ctx, size_t key_len_bytes,
+			  mbedtls_operation_t operation)
+{
+	return mbedtls_cipher_setkey(p_ctx, m_aes_key_buf, key_len_bytes * 8,
+				     operation);
+}
+
+static int cipher_set_padding(mbedtls_cipher_context_t *ctx,
+			      mbedtls_cipher_padding_t padding)
+{
+	/* Only CBC mode supports padding at the moment. */
+	return 0;
+}
+
+static int cipher_update_iterative(mbedtls_cipher_context_t *p_ctx,
+				   size_t input_len)
+{
+	int err_code;
+	size_t round = 0;
+
+	do {
+		size_t process_len = MIN(16, input_len);
+		err_code = mbedtls_cipher_cmac_update(
+			p_ctx, &m_aes_input_buf[16 * round], process_len);
+		input_len -= process_len;
+		round++;
+	} while ((err_code == 0) && (input_len > 0));
+
+	return err_code;
+}
+
+__attribute__((noinline)) static void unhexify_aes(void)
+{
+	bool encrypt = p_test_vector->direction == MBEDTLS_ENCRYPT;
+
+	key_len = hex2bin(p_test_vector->p_key, strlen(p_test_vector->p_key),
+			  m_aes_key_buf, strlen(p_test_vector->p_key));
+	iv_len = hex2bin(p_test_vector->p_iv, strlen(p_test_vector->p_iv),
+			 m_aes_iv_buf, strlen(p_test_vector->p_iv));
+	ad_len = hex2bin(p_test_vector->p_ad, strlen(p_test_vector->p_ad),
+			 m_aes_temp_buf, strlen(p_test_vector->p_ad));
+
+	if (encrypt) {
+		input_len = hex2bin(p_test_vector->p_plaintext,
+				    strlen(p_test_vector->p_plaintext),
+				    m_aes_input_buf,
+				    strlen(p_test_vector->p_plaintext));
+		output_len = hex2bin(p_test_vector->p_ciphertext,
+				     strlen(p_test_vector->p_ciphertext),
+				     m_aes_expected_output_buf,
+				     strlen(p_test_vector->p_ciphertext));
+	} else {
+		input_len = hex2bin(p_test_vector->p_ciphertext,
+				    strlen(p_test_vector->p_ciphertext),
+				    m_aes_input_buf,
+				    strlen(p_test_vector->p_ciphertext));
+		output_len = hex2bin(p_test_vector->p_plaintext,
+				     strlen(p_test_vector->p_plaintext),
+				     m_aes_expected_output_buf,
+				     strlen(p_test_vector->p_plaintext));
+	}
+}
+
+static void aes_setup_ecb_mac(void)
+{
+	aes_ecb_mac_clear_buffers();
+
+	static int i;
+	p_test_vector =
+		ITEM_GET(test_vector_aes_ecb_mac_data, test_vector_aes_t, i++);
+
+	unhexify_aes();
+}
+
+/**@brief Function for the AES MAC test execution.
+ */
+void exec_test_case_aes_ecb_mac(void)
+{
+	int err_code = -1;
+
+	mbedtls_cipher_context_t ctx;
+
+	err_code = cipher_init(&ctx, key_len, p_test_vector->mode);
+	TEST_VECTOR_ASSERT_EQUAL(0, err_code);
+
+	err_code = cipher_set_key(&ctx, key_len, MBEDTLS_ENCRYPT);
+	TEST_VECTOR_ASSERT_EQUAL(0, err_code);
+
+	err_code = cipher_set_padding(&ctx, p_test_vector->padding);
+	TEST_VECTOR_ASSERT_EQUAL(0, err_code);
+
+	start_time_measurement();
+	LOG_DBG("Using ECB for MAC (CMAC).");
```

Comment:
Log should be before start_time_measurement()

Reply:
Moved in 
a69739b